redbot/cogs/audio/audio_dataclasses.py

```diff
@@ -367,18 +448,20 @@ def _parse(track, **kwargs):
             try:
                 query_url = urlparse(track)
                 if all([query_url.scheme, query_url.netloc, query_url.path]):
+                    returning["url"] = track
+                    returning["is_url"] = True
                     url_domain = ".".join(query_url.netloc.split(".")[-2:])
                     if not query_url.netloc:
                         url_domain = ".".join(query_url.path.split("/")[0].split(".")[-2:])
                     if url_domain in ["youtube.com", "youtu.be"]:
                         returning["youtube"] = True
                         _has_index = "&index=" in track
                         if "&t=" in track:
-                            match = re.search(_re_youtube_timestamp, track)
+                            match = re.search(_RE_YOUTUBE_TIMESTAMP, track)
                             if match:
                                 returning["start_time"] = int(match.group(1))
                         if _has_index:
-                            match = re.search(_re_youtube_index, track)
+                            match = re.search(_RE_YOUTUBE_INDEX, track)
                             if match:
                                 returning["track_index"] = int(match.group(1)) - 1
                         if all(k in track for k in ["&list=", "watch?"]):
```

Comment:
(Note for later, not necessary to change in this PR, just noticed while reading through the full context)

The logic from here down including all branches attached to this if statement gets convoluted. There are 2 branches with identical implementations, separated by another branch. It's not valid to merge the branches directly either, and it won't be immediately obvious as to why. This should probably be looked at for a clarity improvement later on.